src/Command/General/TownNewsMigrator.php

```diff
@@ -696,7 +698,7 @@ private function set_post_author( $post_id, $display_name, $author_meta, $defaul
 			);
 		} else {
 			// Set as a co-author.
-			$guest_author = $this->coauthorsplus_logic->get_guest_author_by_user_login( $email );
+			$guest_author = $this->coauthorsplus_logic->get_guest_author_by_user_login( $display_name );
```

Comment:
Using `$email` was a bug here, nice catch! The [create_guest_author](https://github.com/Automattic/newspack-custom-content-migrator/blob/ea5e7a66d1c0727230936284059fbd9e68657519/src/Logic/CoAuthorPlus.php#L105) used [here](https://github.com/Automattic/newspack-custom-content-migrator/pull/374/files#diff-a630c53224b300be80052c56980d3a5f4a5d0ed96e9764671318e1fa9010aaa7R714) to create a new co-author uses the sanitized display_name as user_login in case we don't provide the later, so I think to fetch co-authors using `$display_name` as user_login we should pass the sanitized version.
```suggestion
			$guest_author = $this->coauthorsplus_logic->get_guest_author_by_user_login( sanitize_title( $display_name ) );
```

Reply:
Thanks Zak - good catch! 

I added sanitization with sanitize_user because I assume that is what the user login is run through when it's created. Does that make sense?

Reply:
The user is created as of now with [`sanitize_title`](https://github.com/Automattic/newspack-custom-content-migrator/blob/ea5e7a66d1c0727230936284059fbd9e68657519/src/Logic/CoAuthorPlus.php#L112), but I think it's better to fix it properly and change it there to `sanitize_user`